drivers/rf-transceiver/hmc630x/iio_hmc630x.c

```diff
@@ -0,0 +1,378 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <math.h>
+#include "no_os_error.h"
+#include "no_os_delay.h"
+#include "iio.h"
+#include "iio_hmc630x.h"
+#include "no_os_util.h"
+#include "hmc630x.h"
+#include <string.h>
+
+static int32_t _hmc630x_read_register2(struct hmc630x_iio_dev *iiodev,
+				       uint32_t reg,
+				       uint32_t *readval)
+{
+	int ret;
+	uint8_t val;
+	ret = hmc630x_read_row(iiodev->dev, reg, &val);
```

Comment:
`readval` may be directly passed to this function.

Reply:
i don't think so, behavior is different based on endianess